src/MOCKernel.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file MOCKernel.h
+ * @brief An MOCKernel object
+ * @date May 5, 2015
+ * @author Geoffrey Gunow, MIT, Course 22 (geogunow@mit.edu)
+ */
+
+#ifndef MOCKERNEL_H_
+#define MOCKERNEL_H_
+
+#include "Python.h"
+#include "Track.h"
+#include "Geometry.h"
+#include "Quadrature.h"
+
+
+/* Forward declaration of TrackGenerator */
+class TrackGenerator;
+
+
+/**
+ * @class MOCKernel MOCKernel.h "src/MOCKernel.h"
+ * @brief An MOCKernel object specifies a functionality to apply to MOC
+ *        segments.
+ * @details An MOCKernel is an object that owns some specified data and
+ *          contains an "execute" function which applies some functionality
+ *          to the data. This is useful in MOC where it is very common to
+ *          apply some function to segment data in either a nested loop
+ *          structure or from on-the-fly calculations. Kernels specify the
+ *          actions applied to the segments, reducing the need for repeated
+ *          code. This class is the parent class of CounterKernel and
+ *          VolumeKernel. A generic MOCKernel should not be explicity
+ *          instantiated. Instead, an inhereting class should
+ *          be instantiated which describes the "execute" function.
+ */
+class MOCKernel {
+
+protected:
+
+  /** Count referring to the segment number */
```

Comment:
The definition of ``_count`` is always the same. It is the number of segments traversed since the last reset.